consensus/src/marshal/ingress/mailbox.rs

```diff
@@ -89,6 +104,45 @@ impl<V: Variant, B: Block> Mailbox<V, B> {
         rx
     }
 
+    /// GetByHeight is a request to retrieve a finalized block by its height.
+    /// If the block is not found locally, the response will be [None].
+    pub async fn get_by_height(&mut self, height: u64) -> oneshot::Receiver<Option<B>> {
```

Comment:
Got a design question for you guys:

Why did you opt to make this return a channel rather than make this implementation detail opaque? I am thinking of awaiting the channel instead:

```rust
pub async fn_get_by_height(&mut self, height: u64) -> eyre::Result<Option<B>> {
    let (tx, rx) = oneshot::channel();
    self.sender.send(Message::GetByHeight {
        height,
        response: tx,
    }).await?;
    rx.await
}
```

This way you hide how you communicate internally and have the user avoid this double await, i.e. `mailbox.get_by_height.await.await`